File: LeanStorageDemoObjC/LeanStorageDemo/Student.m


#import "Student.h"

@implementation Student

//重要, 使编译器动态生成getter和setter方法
#warning 为了引起你的注意! 如果明白了用法可以删除这行
@dynamic name, age, gender, hobbies, avatar, any;

+ (NSString *)parseClassName {
    return @"Student";
}

@end

File: LeanStorageDemoSwift/LeanStorageDemoSwift/Student.swift
//
//  Student.swift
//  LeanStorageDemoSwift
//
//  Created by lzw on 15/9/3.
//  Copyright (c) 2015年 微信:  lzwjava. All rights reserved.
//

import UIKit
import AVOSCloud

let kStudentKeyHobbies: String = "hobbies"
let kStudnetKeyAvatar: String = "avatar"
let kStudnetKeyAge: String = "age"
let kStudentKeyAny: String = "any"
let kStudentKeyName: String = "name"
let kStudnetKeyFriends: String = "friends"

class Student: AVObject, AVSubclassing {
    
    @NSManaged var avatar: AVFile?
    
    @NSManaged var name: String?
    
    @NSManaged var age: Int32
    
    @NSManaged var hobbies: [AnyObject]?
    
    @NSManaged var any: AnyObject?
    
    static func parseClassName() -> String! {
        return "Student"
    }
}
